{
 "cells": [
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "    ### colors(color):\n",
    "    \"\"\"\n",
    "    Function : Colors printed text with black,red,green,yellow,blue,magenta,cyan or white\n",
    "    \n",
    "    \"\"\"\n",
    "    black = lambda text: '\\033[0;30m' + text + '\\033[0m'\n",
    "    red = lambda text: '\\033[0;31m' + text + '\\033[0m'\n",
    "    green = lambda text: '\\033[0;32m' + text + '\\033[0m'\n",
    "    yellow = lambda text: '\\033[0;33m' + text + '\\033[0m'\n",
    "    blue = lambda text: '\\033[0;34m' + text + '\\033[0m'\n",
    "    magenta = lambda text: '\\033[0;35m' + text + '\\033[0m'\n",
    "    cyan = lambda text: '\\033[0;36m' + text + '\\033[0m'\n",
    "    white = lambda text: '\\033[0;37m' + text + '\\033[0m'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [],
   "source": [
    "# open_pdb(pdb_file)\n",
    "\n",
    "def menu():\n",
    "    \"\"\"\n",
    "    Function: Displays the menu with different options for the user\n",
    "    Arguments: None\n",
    "    Returns: Menu\n",
    "    \n",
    "    \"\"\"\n",
    "    blue = lambda text: '\\033[0;34m' + text + '\\033[0m' # Colors the printed output blue\n",
    "    red = lambda text: '\\033[0;31m' + text + '\\033[0m'\n",
    "    software = \"PDB FILE ANALYZER\"\n",
    "    choices = \"Select an option from below:\"\n",
    "    c1 = \"1) Open a PDB file                     (O)\"\n",
    "    c2 = \"2) Information                         (I)\"\n",
    "    c3 = \"3) Show histrogram of amino acids      (H)\"\n",
    "    c4 = \"4) Display Secondary Structure         (S)\"\n",
    "    c5 = \"5) Export PDB File                     (X)\"\n",
    "    c6 = \"6) Exit                                (Q)\"\n",
    "    global file\n",
    "    status = \"Current PDB: \"+ file\n",
    "    len(software)\n",
    "\n",
    "    stars=\"*\"\n",
    "    space=\" \"\n",
    "    print(blue(stars*80))\n",
    "    #The length of the inserted string and its 0 index is subtracted for the desired 80 characters per line to be met.\n",
    "    #Subtract index 0s in each object added, this concept is applied to each line.\n",
    "    print(blue(stars * 1),blue(\"%0s\"%\"\"),blue(\"%s\"%software),space*(75-(len(software)))+blue(stars))\n",
    "    print(blue(stars*80))\n",
    "    print(blue(stars * 1),\"%0s\"%\"\",blue(\"%s\" %choices),space*(75-(len(choices)))+blue(stars) ) \n",
    "    print(blue(stars*1), blue(space*76), blue(stars*1))\n",
    "    print(blue(stars * 1),\"%5s\"%\"\",blue(\"%s\" %c1),space*(70-(len(c1)))+blue(stars))\n",
    "    print(blue(stars * 1),\"%5s\"%\"\",blue(\"%s\"%c2),space*(70-(len(c2)))+blue(stars))\n",
    "    print(blue(stars * 1),\"%5s\"%\"\",blue(\"%s\" %c3),space*(70-(len(c3)))+blue(stars))\n",
    "    print(blue(stars * 1),\"%5s\"%\"\",blue(\"%s\" %c4),space*(70-(len(c4)))+blue(stars))\n",
    "    print(blue(stars * 1),\"%5s\"%\"\",blue(\"%s\" %c5),space*(70-(len(c5)))+blue(stars))\n",
    "    print(blue(stars * 1),\"%5s\"%\"\",blue(\"%s\" %c6),space*(70-(len(c6)))+blue(stars))\n",
    "    print(blue(stars * 1),\"%0s\"%\"\",space*(74-(len(status))),blue(\"%s\" %status),blue(stars*1))\n",
    "    print(blue(stars*80))\n",
    "    global choice\n",
    "    choice = input(\":\")     "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fungua(filename):\n",
    "    \"\"\"\n",
    "    Function: Tests a valid file path by loading it to the memory \n",
    "              Test if the file is a pdb file\n",
    "    Argument: Name or valid path of the file\n",
    "    \n",
    "    \"\"\"\n",
    "    with open(filename, 'r') as f:\n",
    "        global file \n",
    "        file = f.name\n",
    "        checkList= ['HEADER','OBSLTE','TITLE','SPLT','CAVEAT','COMPND','SOURCE','KEYWDS','EXPDTA','NUMMDL','MDLTYP',\\\n",
    "    'AUTHOR','REVDAT','SPRSDE','JRNL','REMARKS','DBREF','DBREF1','DBREF2','SEQADV','SEQRES','MODRES',\\\n",
    "    'HET','FORMUL','HETNAM','HETSYN','HELIX','SHEET','SSBOND','LINK','CISPEP','SITE','CRYST1','MTRIXn',\\\n",
    "    'ORIGXn','SCALEn','MODEL','ATOM','ANISOU','TER','HETATM','ENDMDL','CONECT','MASTER','END']\n",
    "        checkList = ','.join(checkList)\n",
    "        for line in f:\n",
    "            if len(line) == 81:\n",
    "                pass\n",
    "                starts = str(line[:6])\n",
    "                status = True\n",
    "                for i in starts:\n",
    "                    if i in checkList:\n",
    "                        pass\n",
    "                    else:\n",
    "                        status = False\n",
    "                return status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [],
   "source": [
    "def choiceO():\n",
    "    global load_file\n",
    "    load_file = str(input('Enter a Valid PATH for a PDB File:'))\n",
    "    if fungua(load_file) == True:\n",
    "        print(\"The File %s has been sucessfully loaded\" %load_file)\n",
    "    else:\n",
    "        print(\"Ivalid file loaded\")\n",
    "        global file\n",
    "        file = \"None\"\n",
    "    menu()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [],
   "source": [
    "def choiceI():\n",
    "    titlePdb() # prints the file name and the title of th pdb file\n",
    "    printChains(load_file)\n",
    "    \n",
    "#     import textwrap\n",
    "\n",
    "#     aa_dic = {'A':'ALA', 'R':'ARG', 'N':'ASN', 'D':'ASP', 'C':'GLY', 'Q':'GLN', 'E':'GLU', 'H':'HIS',\\\n",
    "#               'I':'ILE', 'L':'LEU', 'K':'LYS', 'M':'MET', 'F':'PHE', 'P':'PRO', 'S':'SER', 'T':'THR', 'W':'TRP', 'Y':'TYR', 'V':'VAL'} # dictionary for amino acids\n",
    "#     new_dict = dict([(value, key) for (key, value) in aa_dic.items()]) # swapped the values and keys \n",
    "\n",
    "#     with open(\"3AYU1.pdb\", 'r') as f:\n",
    "#         seqA = \"\"\n",
    "#         seqB = \"\"\n",
    "#         seqC = \"\"\n",
    "#         seqD = \"\"\n",
    "#         for line in f:\n",
    "#             if line.startswith('SEQRES'):\n",
    "#                 l = line.split()[2:]         # creaeting a list of our three letter code amino acids and displays starting from the chain type\n",
    "#                 if l[0] == 'A':\n",
    "#                     l = l[2:]\n",
    "#                     c = [new_dict[codon]for codon in l]\n",
    "#                     seqA += \"\".join(c)\n",
    "\n",
    "#                 elif l[0] == 'B':\n",
    "#                     l = l[2:]\n",
    "#                     c = [new_dict[codon]for codon in l]\n",
    "#                     seqB += \"\".join(c)\n",
    "                \n",
    "#                 elif l[0] == 'C':\n",
    "#                     l = l[2:]\n",
    "#                     c = [new_dict[codon]for codon in l]\n",
    "#                     seqC += \"\".join(c)\n",
    "                \n",
    "#                 elif l[0] == 'D':\n",
    "#                     l = l[2:]\n",
    "#                     c = [new_dict[codon]for codon in l]\n",
    "#                     seqD += \"\".join(c)\n",
    "\n",
    "\n",
    "#         sA = \"\\n\".join(textwrap.wrap(seqA,50))\n",
    "#         sB = \"\\n\".join(textwrap.wrap(seqB,50))\n",
    "#         sC = \"\\n\".join(textwrap.wrap(seqC,50))\n",
    "#         sD = \"\\n\".join(textwrap.wrap(seqD,50))\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [],
   "source": [
    "def titlePdb():\n",
    "    \"\"\"\n",
    "    Function: Extracts the title from the pdb file.\n",
    "    \n",
    "    \"\"\"\n",
    "    red = lambda text: '\\033[0;31m' + text + '\\033[0m'\n",
    "    file_name = load_file\n",
    "    if fungua(file_name):\n",
    "        myLis = []\n",
    "        with open(load_file, 'r') as f:\n",
    "            global file \n",
    "            file = f.name\n",
    "            print(\"PDB File: %s \" %red(file) )\n",
    "            Title = \"\"\n",
    "            for line in f:\n",
    "                if line.startswith('TITLE'):\n",
    "                    Title = line.strip('TITLE')\n",
    "                    Title = Title.strip()\n",
    "                    myLis.append(Title)\n",
    "            myString = str((\"\").join(myLis)) # joining the list and converting it into a string\n",
    "            myString = \"Title: \" + myString.strip()\n",
    "            if len(myString) <= 80:\n",
    "                print(myString[:80])\n",
    "            else:\n",
    "                print(myString[:80]+\"\\n\"+myString[80:])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chains: AB and P\n"
     ]
    }
   ],
   "source": [
    "def printChains(load_file):\n",
    "    \"\"\" \n",
    "    Input: pdb file\n",
    "    Function: Prints all the chains in the pdb file\n",
    "    \"\"\"\n",
    "    with open(load_file, 'r') as f:\n",
    "        chain = list(\"ABCDEFGHIJKLMNOPQRSTUVWXYZ\")\n",
    "        lyst = \"\"\n",
    "        for line in f:\n",
    "            if line.startswith('SEQRES'):\n",
    "                s = line.split()[2:]         # creaeting a list of our three letter code amino acids and displays starting from the chain typ\n",
    "                for i in s:\n",
    "                    for letter in chain:\n",
    "                        if i == letter:\n",
    "                            lyst+= letter\n",
    "        lyst = sorted(lyst)             #sort the list\n",
    "        lyst = list(dict.fromkeys(lyst)) # remove duplicates leaving us with the number of chains in the pdb file\n",
    "        lyst = \"\".join(lyst)\n",
    "        global chainString\n",
    "        chainString = str(lyst[:-1])+str(lyst[-1])\n",
    "        allChains = str(lyst[:-1])+' '+'and'+' '+str(lyst[-1])\n",
    "        print('Chains: '+ allChains)\n",
    "printChains(load_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ABP\n",
      " - Chain A\n",
      "     Number of amino acids:  167\n",
      "     Number of helix: %9s\n",
      "     Number of sheet: %9s\n",
      "     Sequence: \n"
     ]
    }
   ],
   "source": [
    "    \n",
    "    import textwrap # introduces prints stirng of specified length\n",
    "\n",
    "    aa_dic = {'A':'ALA', 'R':'ARG', 'N':'ASN', 'D':'ASP', 'C':'GLY', 'Q':'GLN', 'E':'GLU', 'H':'HIS',\\\n",
    "              'I':'ILE', 'L':'LEU', 'K':'LYS', 'M':'MET', 'F':'PHE', 'P':'PRO', 'S':'SER', 'T':'THR', 'W':'TRP', 'Y':'TYR', 'V':'VAL'} # dictionary for amino acids\n",
    "    new_dict = dict([(value, key) for (key, value) in aa_dic.items()]) # swapped the values and keys \n",
    "\n",
    "    with open(\"3AYU1.pdb\", 'r') as f:\n",
    "        seqA = \"\"\n",
    "        seqB = \"\"\n",
    "        seqC = \"\"\n",
    "        seqD = \"\"\n",
    "        seqP = \"\"\n",
    "        for line in f:\n",
    "            if line.startswith('SEQRES'):\n",
    "                l = line.split()[2:]         # creaeting a list of our three letter code amino acids and displays starting from the chain type\n",
    "                if l[0] == 'A':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqA += \"\".join(c)\n",
    "\n",
    "                elif l[0] == 'B':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqB += \"\".join(c)\n",
    "                \n",
    "                elif l[0] == 'C':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqC += \"\".join(c)\n",
    "                \n",
    "                elif l[0] == 'D':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqD += \"\".join(c)\n",
    "                    \n",
    "                elif l[0] == 'P':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqP += \"\".join(c)\n",
    "\n",
    "\n",
    "        sA = \"\\n\".join(textwrap.wrap(seqA,50))\n",
    "        sB = \"\\n\".join(textwrap.wrap(seqB,50))\n",
    "        sC = \"\\n\".join(textwrap.wrap(seqC,50))\n",
    "        sD = \"\\n\".join(textwrap.wrap(seqD,50))\n",
    "        sP = \"\\n\".join(textwrap.wrap(seqP,50))\n",
    "    def no_aa(seq):\n",
    "        return len(seq)-(len(seq)//50)\n",
    "    print(chainString)\n",
    "    for i in chainString:\n",
    "        if i == \"A\":\n",
    "            print(\" - Chain %s\" % i)\n",
    "            print(\"     Number of amino acids: \", no_aa(sA))\n",
    "            print(\"     Number of helix: %9s\")\n",
    "            print(\"     Number of sheet: %9s\")\n",
    "            print(\"     Sequence:\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ABP\n",
      " - Chain A\n",
      "     Number of amino acids:  167\n",
      "     Number of helix: %9s\n",
      "     Number of sheet: %9s\n",
      "     Sequence: YNFFPRKPKWDKNQITYRIICYTPDLDPETVDDAFARAFQVWSDVTPLRF\n",
      "SRIHDCEADIMINFCRWEHCDCYPFDCKDCLLAHAFAPCTCVCCDSHFDD\n",
      "DELWTLCKCVCYSLFLVAAHAFCHAMCLEHSQDPCALMAPIYTYTKNFRL\n",
      "SQDDIKCIQELYCASPD\n",
      " - Chain B\n",
      "     Number of amino acids:  10\n",
      "     Number of helix: %9s\n",
      "     Number of sheet: %9s\n",
      "     Sequence: ISYCNDALMP\n",
      " - Chain P\n",
      "     Number of amino acids:  0\n",
      "     Number of helix: %9s\n",
      "     Number of sheet: %9s\n",
      "     Sequence:                 \n"
     ]
    }
   ],
   "source": [
    "    \n",
    "    import textwrap # introduces prints stirng of specified length\n",
    "\n",
    "    aa_dic = {'A':'ALA', 'R':'ARG', 'N':'ASN', 'D':'ASP', 'C':'GLY', 'Q':'GLN', 'E':'GLU', 'H':'HIS',\\\n",
    "              'I':'ILE', 'L':'LEU', 'K':'LYS', 'M':'MET', 'F':'PHE', 'P':'PRO', 'S':'SER', 'T':'THR', 'W':'TRP', 'Y':'TYR', 'V':'VAL'} # dictionary for amino acids\n",
    "    new_dict = dict([(value, key) for (key, value) in aa_dic.items()]) # swapped the values and keys \n",
    "\n",
    "    with open(\"3AYU1.pdb\", 'r') as f:\n",
    "        seqA = \"\"\n",
    "        seqB = \"\"\n",
    "        seqC = \"\"\n",
    "        seqD = \"\"\n",
    "        seqP = \"\"\n",
    "        for line in f:\n",
    "            if line.startswith('SEQRES'):\n",
    "                l = line.split()[2:]         # creaeting a list of our three letter code amino acids and displays starting from the chain type\n",
    "                if l[0] == 'A':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqA += \"\".join(c)\n",
    "\n",
    "                elif l[0] == 'B':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqB += \"\".join(c)\n",
    "                \n",
    "                elif l[0] == 'C':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqC += \"\".join(c)\n",
    "                \n",
    "                elif l[0] == 'D':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqD += \"\".join(c)\n",
    "                    \n",
    "                elif l[0] == 'P':\n",
    "                    l = l[2:]\n",
    "                    c = [new_dict[codon]for codon in l]\n",
    "                    seqP += \"\".join(c)\n",
    "\n",
    "\n",
    "        sA = \"\\n\".join(textwrap.wrap(seqA,50))\n",
    "        sB = \"\\n\".join(textwrap.wrap(seqB,50))\n",
    "        sC = \"\\n\".join(textwrap.wrap(seqC,50))\n",
    "        sD = \"\\n\".join(textwrap.wrap(seqD,50))\n",
    "        sP = \"\\n\".join(textwrap.wrap(seqP,50))\n",
    "    def no_aa(seq):\n",
    "        return len(seq)-(len(seq)//50)\n",
    "    print(chainString)\n",
    "    for i in chainString:\n",
    "        if i == \"A\":\n",
    "            print(\" - Chain %s\" % i)\n",
    "            print(\"     Number of amino acids: \", no_aa(sA))\n",
    "            print(\"     Number of helix: %9s\")\n",
    "            print(\"     Number of sheet: %9s\")\n",
    "            print(\"     Sequence: %s\" % sA)\n",
    "        elif i == \"B\":\n",
    "            print(\" - Chain %s\" % i)\n",
    "            print(\"     Number of amino acids: \", no_aa(sB))\n",
    "            print(\"     Number of helix: %9s\")\n",
    "            print(\"     Number of sheet: %9s\")\n",
    "            print(\"     Sequence: %s\" % sB)\n",
    "        elif i == \"C\":\n",
    "            print(\" - Chain %s\" % i)\n",
    "            print(\"     Number of amino acids: \", no_aa(sC))\n",
    "            print(\"     Number of helix: %9s\")\n",
    "            print(\"     Number of sheet: %9s\")\n",
    "            print(\"     Sequence: %16s\" % sC)\n",
    "        elif i == \"D\":\n",
    "            print(\" - Chain %s\" % i)\n",
    "            print(\"     Number of amino acids: \", no_aa(sD))\n",
    "            print(\"     Number of helix: %9s\")\n",
    "            print(\"     Number of sheet: %9s\")\n",
    "            print(\"     Sequence: %s\" % sD)\n",
    "        elif i == \"P\":\n",
    "            print(\" - Chain %s\" % i)\n",
    "            print(\"     Number of amino acids: \", no_aa(sP))\n",
    "            print(\"     Number of helix: %9s\")\n",
    "            print(\"     Number of sheet: %9s\")\n",
    "            print(\"     Sequence: %16s\" % sP)\n",
    "\n"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Title: CRYSTAL STRUCTURE OF MMP-2 ACTIVE SITE MUTANT IN COMPLEX WITH APP-DRIVED\n",
    "DECAPEPTIDE INHIBITOR\n",
    "CHAINS: A and B\n",
    "- Chain A\n",
    "Number of amino acids:\n",
    "167\n",
    "Number of helix:\n",
    "3\n",
    "Number of sheet:\n",
    "9\n",
    "Sequence: YNFFPRKPKWDKNQITYRIIGYTPDLDPETVDDAFARAFQVWSDVTPLRF\n",
    "SRIHDGEADIMINFGRWEHGDGYPFDGKDGLLAHAFAPGTGVGGDSHFDD\n",
    "DELWTLGKGVGYSLFLVAAHAFGHAMGLEHSQDPGALMAPIYTYTKNFRL\n",
    "SQDDIKGIQELYGASPD\n",
    "- Chain B\n",
    "Number of amino acids:\n",
    "10\n",
    "Number of helix:\n",
    "0\n",
    "Number of sheet:\n",
    "1\n",
    "Sequence: ISYGNDALMP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [],
   "source": [
    "import textwrap\n",
    "\n",
    "aa_dic = {'A':'ALA', 'R':'ARG', 'N':'ASN', 'D':'ASP', 'C':'GLY', 'Q':'GLN', 'E':'GLU', 'H':'HIS',\\\n",
    "          'I':'ILE', 'L':'LEU', 'K':'LYS', 'M':'MET', 'F':'PHE', 'P':'PRO', 'S':'SER', 'T':'THR', 'W':'TRP', 'Y':'TYR', 'V':'VAL'} # dictionary for amino acids\n",
    "new_dict = dict([(value, key) for (key, value) in aa_dic.items()]) # swapped the values and keys \n",
    "\n",
    "with open(\"3AYU1.pdb\", 'r') as f:\n",
    "    chain = list(\"ABCDEFGHIJKLMNOPQRSTUVWXYZ\")\n",
    "    for line in f:\n",
    "        if line.startswith('SEQRES'):\n",
    "            l = line.split()[2:]\n",
    "            for i in chain:\n",
    "                chainsNames(i,l)\n",
    "                \n",
    "                \n",
    "\n",
    "def chainsNames(letter, l):\n",
    "    \"\"\" \n",
    "    input:     single letter name of the chain \n",
    "    function : Takes the name of the chain e.g A, B, C ... and returns a string of its one letter code amino acid sequence\n",
    "    \"\"\"\n",
    "    seq = \"\"\n",
    "    if l[0] == letter:\n",
    "        l = l[2:]\n",
    "        c = [new_dict[codon]for codon in l]\n",
    "        seq += \"\".join(c)\n",
    "    return seq\n",
    "\n",
    "# def printChain(seq):\n",
    "#     \"\"\"\n",
    "#     Input: a string \n",
    "#     Function : prints 50 characters per line of the string\n",
    "#     \"\"\"\n",
    "#     print(\"\\n\".join(textwrap.wrap(seq,50)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "______"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [],
   "source": [
    "def choices():\n",
    "    \"\"\"\n",
    "    Function: Executes the different options ('o','i','h','s','x' and 'q'): each option is a function of its own.\n",
    "    \n",
    "    \"\"\"\n",
    "    if choice.lower() in ('o', 'i', 'h', 's', 'x', 'q'):\n",
    "            if choice.lower() == 'o':\n",
    "                choiceO()\n",
    "                if choice.lower() == 'i':\n",
    "                    choiceI()                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;34m********************************************************************************\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m \u001b[0;34m\u001b[0m \u001b[0;34mPDB FILE ANALYZER\u001b[0m                                                           \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m********************************************************************************\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m  \u001b[0;34mSelect an option from below:\u001b[0m                                                \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m \u001b[0;34m                                                                            \u001b[0m \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m1) Open a PDB file                     (O)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m2) Information                         (I)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m3) Show histrogram of amino acids      (H)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m4) Display Secondary Structure         (S)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m5) Export PDB File                     (X)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m6) Exit                                (Q)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m                                                            \u001b[0;34mCurrent PDB: None\u001b[0m \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m********************************************************************************\u001b[0m\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      ": o\n",
      "Enter a Valid PATH for a PDB File: ../Data/1kjf.pdb\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The File ../Data/1kjf.pdb has been sucessfully loaded\n",
      "\u001b[0;34m********************************************************************************\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m \u001b[0;34m\u001b[0m \u001b[0;34mPDB FILE ANALYZER\u001b[0m                                                           \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m********************************************************************************\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m  \u001b[0;34mSelect an option from below:\u001b[0m                                                \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m \u001b[0;34m                                                                            \u001b[0m \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m1) Open a PDB file                     (O)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m2) Information                         (I)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m3) Show histrogram of amino acids      (H)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m4) Display Secondary Structure         (S)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m5) Export PDB File                     (X)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m       \u001b[0;34m6) Exit                                (Q)\u001b[0m                             \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m*\u001b[0m                                                \u001b[0;34mCurrent PDB: ../Data/1kjf.pdb\u001b[0m \u001b[0;34m*\u001b[0m\n",
      "\u001b[0;34m********************************************************************************\u001b[0m\n"
     ]
    },
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      ": i\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PDB File: \u001b[0;31m../Data/1kjf.pdb\u001b[0m \n",
      "Title: SUBSTRATE SHAPE DETERMINES SPECIFICITY OF RECOGNITION2 RECOGNITION FOR HI\n",
      "V-1 PROTEASE: ANALYSIS OF CRYSTAL3 STRUCTURES OF SIX SUBSTRATE COMPLEXES\n",
      "Chains: AB and P\n"
     ]
    }
   ],
   "source": [
    "def main1():\n",
    "    global file\n",
    "    file= \"None\"\n",
    "    menu()\n",
    "    \n",
    "    if choice.lower() in ('o', 'i', 'h', 's', 'x', 'q'):\n",
    "        while choice != 'q':\n",
    "            try:\n",
    "                choices()\n",
    "                break\n",
    "            except:\n",
    "                print('Invalid file Loaded')\n",
    "                choices()\n",
    "    else:\n",
    "        print(\"Invalid choice selected\")\n",
    "        main1()\n",
    "\n",
    "main1()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
